src/internal.c

```diff
@@ -4643,11 +4643,20 @@ void FreeX509(WOLFSSL_X509* x509)
         x509->altNames = NULL;
     }
 
-#ifdef WOLFSSL_DUAL_ALG_CERTS
-    XFREE(x509->sapkiDer, x509->heap, DYNAMIC_TYPE_X509_EXT);
-    XFREE(x509->altSigAlgDer, x509->heap, DYNAMIC_TYPE_X509_EXT);
-    XFREE(x509->altSigValDer, x509->heap, DYNAMIC_TYPE_X509_EXT);
-#endif /* WOLFSSL_DUAL_ALG_CERTS */
+    #ifdef WOLFSSL_DUAL_ALG_CERTS
+    if (x509->sapkiDer) {
```

Comment:
FYI: XFREE is supposed to have a NULL check on it. All of the default XFREE implementations do and if a customer implements their own it also needs to. We have a large number of places in the code that make this assumption.